Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -542,6 +543,46 @@ private static void Clean(NugetPackageIdentifier package)
             }
         }
 
+        private static bool IsAlreadyImportedInEngine(NugetPackageIdentifier package) {
+            var alreadyImportedLibs = GetAlreadyImportedLibs();
+            var isAlreadyImported = alreadyImportedLibs.Contains(package.Id);
+            LogVerbose("Is package '{0}' already imported? {1}", package.Id, isAlreadyImported);
+            return isAlreadyImported;
+        }
+
+        private static HashSet<string> GetAlreadyImportedLibs() {
+            var lookupPaths = GetAllLookupPaths();
+            var libNames = lookupPaths
+                .SelectMany(directory => Directory.EnumerateFiles(directory, "*.dll", SearchOption.AllDirectories))
+                .Select(Path.GetFileName)
+                .Select(p => Path.ChangeExtension(p, null));
+            var result = new HashSet<string>(libNames);
```

Comment:
I think we can safely cache this. I'm seeing roughly 1/3 of a second per call to this method, which if there's a big dependency chain, can add up.